lib/wavefront-direct-reporter.js

```diff
@@ -0,0 +1,229 @@
+ * @param {String} token The Wavefront API token with direct ingestion permission.
+ * @param {object} tags The tags to add to wavefront measurements,defaults to none if not specified.
+ * @constructor
+ */
+function WavefrontDirectReporter(registry, prefix, server, token, globaltags, gzip) {
+  WavefrontDirectReporter.super_.call(this, registry);
+  this.prefix = prefix;
+  this.server = server;
+  this.token = token;
+  this.batchSize = 10000;
+  this.gzip = gzip;
+  this.globaltags = {};
+  if (!isEmpty(globaltags)) {
+    this.globaltags = globaltags;
+  }else{
+    this.globaltags = {};
+  }
+}
+
+util.inherits(WavefrontDirectReporter, ScheduledReporter);
+
+WavefrontDirectReporter.prototype.report = function() {
+  var metrics = this.getMetrics();
+  var self = this;
+  var points = [];
+
+  if(metrics.counters.length != 0) {
+    metrics.counters.forEach(function (count) {
+      self.appendCounter.bind(self)(count, points);
+    })
+  }
+
+  if(metrics.meters.length != 0) {
+    metrics.meters.forEach(function (meter) {
+      self.appendMeter.bind(self)(meter, points);
+    })
+  }
+
+  if(metrics.timers.length != 0) {
+    metrics.timers.forEach(function (timer) {
+      // Don't log timer if its recorded no metrics.
+      if(timer.min() != null) {
+        self.appendTimer.bind(self)(timer, points);
+      }
+    })
+  }
+
+  if(metrics.histograms.length != 0) {
+    metrics.histograms.forEach(function (histogram) {
+      // Don't log histogram if its recorded no metrics.
+      if(histogram.min != null) {
+        self.appendHistogram.bind(self)(histogram, points);
+      }
+    })
+  }
+
+  if (points.length != 0) {
+    if (points.length >= self.batchSize) {
+      chunks = getChunks(points, self.batchSize);
+      for (i = 0; i < chunks.length; i++) {
+        self.reportPoints.bind(self)(chunks[i]);
+      }
+    } else {
+      self.reportPoints.bind(self)(points);
+    }
+  }
+};
+
+WavefrontDirectReporter.prototype.reportPoints = function(points) {
+  var self = this;
+  var pointsStr = points.join('\n');
+  var options = {
+    hostname: self.server,
+    path: '/report?f=graphite_v2',
+    method: 'POST',
+    agent: agent,
+    headers: {
+         'Content-Type': 'text/plain',
+         'Authorization': 'Bearer ' + self.token
+    }
+  };
+
+  if (self.gzip) {
+    console.log("using gzip");
+    const buffer = Buffer.from(pointsStr);
+    zlib.gzip(buffer, (err, buffer) => {
+      if (!err) {
+        options.headers['Content-Encoding'] = 'gzip';
+        options.headers['Content-Length'] = buffer.length;
+        self.reportToServer.bind(self)(buffer, options);
+      } else {
+        console.error(err);
+      }
+    });
+  } else {
+    console.log("non-gzip");
+    options.headers['Content-Length'] = pointsStr.length;
+    self.reportToServer.bind(self)(pointsStr, options);
+  }
+};
+
+WavefrontDirectReporter.prototype.reportToServer = function(data, options) {
```

Comment:
Do we ever want to let people know what's going on with the reporter?  There's no way to figure out if the report call failed or not.